### PR TITLE
ENH: Adding transform widget 

### PR DESCRIPTION
**Pull request by adamrankin**
Widget that displays status of VR transform nodes

Click enables/disables Slicer transform updating from VR subsystem

### CODE REVIEW

**Comment by cpinter** [comment]
Thanks! Can you please post a video/gif, or at least a screenshot?

**Comment by adamrankin** [comment]
controllers
out_of_view
normal

File: VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityTransformWidgetPrivate::init()
+{
+  Q_Q(qMRMLVirtualRealityTransformWidget);
+  this->setupUi(q);
+
+  QObject::connect(this->pushButton_Transform, &QPushButton::clicked, q, &qMRMLVirtualRealityTransformWidget::onButtonClicked);
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx]: Just wondering: is this some kind of new way of doing connections in Qt, or just a different way that existed before? (Usually we use SIGNAL(clicked()) and SLOT(onButtonClicked) )

File: VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+void qMRMLVirtualRealityTransformWidget::findViewNode()
+{
+  Q_D(qMRMLVirtualRealityTransformWidget);
+
+  if (d->VRViewNode == nullptr)
+  {
+    // Get access to the VR subsystem
+    for (QWidget* widget : qSlicerApplication::application()->topLevelWidgets())
+    {
+      if (widget->objectName().compare("VirtualRealityWidget") == 0)
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx]: This is a bit of a hack. Is it possible to set it somehow from outside, or make it accessible without traversing widgets and comparing object names?

File: VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+{
+  Q_D(qMRMLVirtualRealityTransformWidget);
+
+  d->TransformNode = node;
+  if (node == nullptr)
+  {
+    return;
+  }
+
+  findViewNode();
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx]: If you have a call like this that has no error check no return value, then it makes the code harder to read. I'd probably set the member variable from the return value of this function and at the same time check the return value and log error if not found.

**Comment by cpinter** [comment]
Thanks, Adam! I made some comments.

I also have a general question. What is the purpose of the qMRMLSequenceBrowserPlayWidgetPlugin? Does it really belong here in the VR extension?

File: VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx (hunk 1 above)
Comment (adamrankin) [reply VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx]: Yes, this it the new C++11 way of doing Qt connections. You can also embed lambdas directly into the SLOT side.

See [https://wiki.qt.io/New_Signal_Slot_Syntax](https://wiki.qt.io/New_Signal_Slot_Syntax)

File: VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx (hunk 2 above)
Comment (adamrankin) [reply VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx]: Definitely, I hate it. But the only other way is to access it via the module class, and you can't access the module library from the widget library.

Comment (adamrankin) [reply VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx]: Maybe the view node could be passed into the widget constructor? That's not bad...

**Comment by adamrankin** [comment]
I don't know what that plugin is. I can't find it in the code.

File: VirtualReality/Widgets/DesignerPlugins/qMRMLSequenceBrowserPlayWidgetPlugin.cxx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+==============================================================================*/
+
+#include "qMRMLSequenceBrowserPlayWidgetPlugin.h"
+#include "qMRMLSequenceBrowserPlayWidget.h"
+
+//------------------------------------------------------------------------------
```
Comment (jcfr) [inline VirtualReality/Widgets/DesignerPlugins/qMRMLSequenceBrowserPlayWidgetPlugin.cxx]: @adamrankin  Here is the file @cpinter was referencing in his comment

**Comment by jcfr** [comment]
Thanks for working on this :pray: 

Would it be possible to add a section to the README file describing the different transform, and what it means to enable/disable each check box ?

See https://github.com/KitwareMedical/SlicerVirtualReality#readme

File: VirtualReality/Widgets/DesignerPlugins/qMRMLSequenceBrowserPlayWidgetPlugin.cxx (hunk 4 above)
Comment (cpinter) [reply VirtualReality/Widgets/DesignerPlugins/qMRMLSequenceBrowserPlayWidgetPlugin.cxx]: Yes I meant anything containing the word "Sequence" that is included in this PR.

Comment (adamrankin) [reply VirtualReality/Widgets/DesignerPlugins/qMRMLSequenceBrowserPlayWidgetPlugin.cxx]: I... I didn't see this. I have no idea how this got added in.

**Comment by cpinter** [comment]
Great, thanks for removing the sequence related files!
I think all my comments have now been adressed, so as far as I'm concerned, it's good to go! Thanks again for working on this!

**Pull request merged**